src/sqlfluff/rules/L011.py

```diff
@@ -95,17 +95,17 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
 
                     # Add initial whitespace if we need to...
                     assert context.raw_segment_pre
-                    if context.raw_segment_pre.name not in ["whitespace", "newline"]:
+                    if not context.raw_segment_pre.is_type("whitespace", "newline"):
                         insert_buff.append(WhitespaceSegment())
 
                     # Add an AS (Uppercase for now, but could be corrected later)
                     insert_buff.append(KeywordSegment("AS"))
 
                     # Add a trailing whitespace if we need to
-                    if context.segment.segments[0].name not in [
+                    if not context.segment.segments[0].is_type(
```

Comment:
Now that we're using types, I'm curious is `.is_type("whitespace")` would also work here. I.e., does `newline` type inherit from `whitespace` type?

Reply:
It's an excellent point - however for some reason I found this note in the `NewlineSegment` class:

```
    NOTE: NewlineSegment does not inherit from WhitespaceSegment.
    Therefore NewlineSegment.is_type('whitespace') returns False.

    This is intentional and convenient for rules. If users want
    to match on both, call .is_type('whitespace', 'newline')
```

Maybe I should revisit that - but for now this seems intentional.